debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/RecordsSnapshotProducerIT.java

```diff
@@ -480,6 +487,118 @@ public void shouldNotSnapshotMaterializedViews() throws Exception {
+
+        buildNoStreamProducer(TestHelper.defaultConfig()
+                .with(PostgresConnectorConfig.DECIMAL_HANDLING_MODE, DecimalHandlingMode.DOUBLE)
+                .with(PostgresConnectorConfig.INCLUDE_UNKNOWN_DATATYPES, true)
+                .with("column.propagate.source.type", "public.alias_table.area,public.alias_table.a"));
+
+        final TestConsumer consumer = testConsumer(1, "public");
+        consumer.await(TestHelper.waitTimeForRecords() * 30, TimeUnit.SECONDS);
+
+        // Specifying alias money2 results in JDBC type '2001' for 'salary2'
+        // Specifying money results in JDBC type '8' for 'salary'
+
+        List<SchemaAndValueField> expected = Arrays.asList(
+                new SchemaAndValueField("salary", Decimal.builder(2).optional().build(), BigDecimal.valueOf(7.25)),
+                new SchemaAndValueField("salary2", Decimal.builder(2).optional().build(), BigDecimal.valueOf(8.25)),
+                new SchemaAndValueField("a", SchemaBuilder.float64().optional()
+                        .parameter(TestHelper.TYPE_NAME_PARAMETER_KEY, "NUMERIC")
+                        .parameter(TestHelper.TYPE_LENGTH_PARAMETER_KEY, "8")
+                        .parameter(TestHelper.TYPE_SCALE_PARAMETER_KEY, "3")
+                        .build(), 12345.123),
+                new SchemaAndValueField("area", SchemaBuilder.float64().optional()
+                        .parameter(TestHelper.TYPE_NAME_PARAMETER_KEY, "FLOAT83")
+                        .parameter(TestHelper.TYPE_LENGTH_PARAMETER_KEY, "8")
+                        .parameter(TestHelper.TYPE_SCALE_PARAMETER_KEY, "3")
+                        .build(), 12345.123));
+
+        consumer.process(record -> assertReadRecord(record, Collect.hashMapOf("public.alias_table", expected)));
+    }
+
+    @Test
+    @FixFor("DBZ-1413")
+    public void shouldSnapshotDomainAliasWithProperModifiers() throws Exception {
+        TestHelper.dropAllSchemas();
+        TestHelper.execute("CREATE DOMAIN varbit2 AS varbit(3);");
+        TestHelper.execute("CREATE TABLE alias_table (pk SERIAL, value varbit2 NOT NULL, PRIMARY KEY(pk));");
+        TestHelper.execute("INSERT INTO alias_table (value) values (B'101');");
+
+        buildNoStreamProducer(TestHelper.defaultConfig()
+                .with(PostgresConnectorConfig.DECIMAL_HANDLING_MODE, DecimalHandlingMode.DOUBLE)
+                .with(PostgresConnectorConfig.INCLUDE_UNKNOWN_DATATYPES, true));
+
+        final TestConsumer consumer = testConsumer(1, "public");
+        consumer.await(TestHelper.waitTimeForRecords() * 30, TimeUnit.SECONDS);
+
+        List<SchemaAndValueField> expected = Collections.singletonList(
+                new SchemaAndValueField("value", Bits.builder(3).build(), new byte[]{ 5, 0 }));
+
+        consumer.process(record -> assertReadRecord(record, Collect.hashMapOf("public.alias_table", expected)));
+    }
+
+    @Test(expected = ComparisonFailure.class)
```

Comment:
Interesting approach to expect this test failure.

Reply:
I mainly added this to catch the possibility in the future where the driver's type modifier support for nested domain types is improved to support them properly, this test would pass rather than fail indicating we needed to revisit this and update documentation/support accordingly.

Reply:
Yepp, that's cool.